### DISCUSSION TITLE
Multiple checkboxes get squashed into one value in the update content on create

### DISCUSSION THREAD

**Discussion by erikbml** [filamentphp/filament#4867] [Help] [answered]
### Package

filament/filament

### Package Version

v2.16.41

### Laravel Version

v.9.38.0

### PHP Version

PHP 8.0.22

### Problem description

I have a custom field, in which I have a group of checkboxes. When this field is used in the CREATE logic, filament only submits the last of the checked checkbox values as a string. When this field is in the EDIT logic, it works correctly. Then I get an array of options in PHP.

### Expected behavior

I expect the create flow to work the same way as the edit flow works; honoring the values selected and not just post the last value

### Steps to reproduce

1. git clone the project
2. composer install
3. create a .env and do php artisan key:generate
4. php artisan migrate
5. php artisan make:filament-user
6. login to /admin
7. go to questionnaire invites module
8. create an invite, with more than one checkbox selected

This will give an error about a string where an array is expected. If you look at the database, you can see that the entry has a value like this:

`{"for_departments":"benchmarking"}`

But it should be like this:

`{"for_departments":["benchmarking"]}`

If you change the value to be like that, with an array and you edit the invite, you can then add extra checkboxes, save it, and it will all work as expected

### Reproduction repository

https://github.com/erikbml/filament-checkboxes-bug

**Comment by danharrin** [comment, marked as answer]
I bet that you just need `default([])` like we have on our own CheckboxList component: https://github.com/filamentphp/filament/blob/2.x/packages/forms/src/Components/CheckboxList.php#L28

We don't really debug issues with specific custom fields here as they aren't really bugs in Filament, you're best asking in GitHub discussions.

**Reply by erikbml** [reply]
Using this setUp technique did not work for me, but converting it to a CheckboxList did work - so thanks!

**Reply by danharrin** [reply]
You could always compare changes between the two. It's probably something very small...
